### DISCUSSION TITLE
Is func.date only for non-datetime field ranges in SQLAlchemy?

### DISCUSSION THREAD

**Discussion by kurianbenoy-aot** [sqlalchemy/sqlalchemy#7467] [Usage Questions] [answered]
In a postgresql database which I have with the name Application with field

created - DataTime Field
modified - Datetime field

When I am querying with:

```
db.session.query(
                Application.application_name,
                Application.application_status,
                func.count(Application.application_name).label("count"),
            )
            .filter(
                and_(
                    func.date(Application.modified) >= from_date,
                    func.date(Application.modified) <= to_date,
                    Application.form_process_mapper_id == mapper_id,
                )
            )
```

I am noticing the results returned are not correct: 

when I am passing the values: from_date =2021-11-30T18:30:00%2B00:00 and to_date=2021-12-01T18:29:59%2B00:00

While same query works in case of:

```
db.session.query(
                Application.application_name,
                Application.application_status,
                func.count(Application.application_name).label("count"),
            )
            .filter(
                and_(
                    Application.modified >= from_date,
                    Application.modified <= to_date,
                    Application.form_process_mapper_id == mapper_id,
                )
            )
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

`func.date` just renders `date()` in the sql. If that's should be used or not depends on what you are trying to do, sqlalchemy has no say in that

**Reply by kurianbenoy-aot** [reply]
So @CaselIT in case of datetime fields does sqlalchemy accept iso-formatted strings, or should it be in python's datetime object. Thanks for clarifying `func.date` is equivalent of date()

**Reply by CaselIT** [reply]
I think for postgresql there is no special logic, so a string is passed to the db. Postgresql will accept an iso formatted string as date/datetime

**Reply by kurianbenoy-aot** [reply]
Ok. Thanks for clarifying
